Libraries/test/Amazon.Lambda.Annotations.SourceGenerators.Tests/WriterTests/ProjectFileHandlerTests.cs

```diff
@@ -96,5 +97,143 @@ public void IsNotOptedOut_PresentButFalse()
+
+                Assert.False(result);
+            }
+            finally
+            {
+                File.Delete(tempFile);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that parsing a csproj with a single TargetFramework trumps the plural TargetFrameworks
+        /// </summary>
+        [Fact]
+        public void TryDetermineTargetFramework_SingleAndMultiple_Success()
+        {
+            var tempFile = Path.GetTempFileName();
+            var csprojContent = @"<Project Sdk=""Microsoft.NET.Sdk"">
+                                    <PropertyGroup>
+                                        <TargetFramework>net8.0</TargetFramework>
+                                        <TargetFrameworks>net7.0</TargetFrameworks>
+                                    </PropertyGroup>
+                                </Project>";
+
+            try 
+            { 
+                File.WriteAllText(tempFile, csprojContent);
+
+                var result = ProjectFileHandler.TryDetermineTargetFramework(tempFile, out var targetFramework);
+
+                Assert.True(result);
+                Assert.Equal("net8.0", targetFramework); // "TargetFramework" should trump "TargetFrameworks
+            }
+            finally
+            {
+                File.Delete(tempFile);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that parsing a csproj with a single value in the plural TargetFrameworks property works
+        /// </summary>
+        [Fact]
+        public void TryDetermineTargetFramework_Multiple_Success()
+        {
+            var tempFile = Path.GetTempFileName();
+            var csprojContent = @"<Project Sdk=""Microsoft.NET.Sdk"">
+                                    <PropertyGroup>
+                                        <TargetFrameworks>net7.0</TargetFrameworks>
+                                    </PropertyGroup>
+                                </Project>";
+
+            try
+            {
+                File.WriteAllText(tempFile, csprojContent);
+
+                var result = ProjectFileHandler.TryDetermineTargetFramework(tempFile, out var targetFramework);
+
+                Assert.True(result);
+                Assert.Equal("net7.0", targetFramework);
+            }
+            finally
+            {
+                File.Delete(tempFile);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that parsing a csproj with a multiple values in the plural TargetFrameworks property fails
+        /// </summary>
+        [Fact]
+        public void TryDetermineTargetFramework_Multiple_Failure()
+        {
+            var tempFile = Path.GetTempFileName();
+            var csprojContent = @"<Project Sdk=""Microsoft.NET.Sdk"">
+                                    <PropertyGroup>
+                                        <TargetFrameworks>net6.0;net8.0</TargetFrameworks>
```

Comment:
Is this a use case we'd want to support? Can't we just pick the minimum supported version?

Reply:
For plural `TargetFrameworks`, I believe we rely on the user specifying the one they actually intend to publish via something like `dotnet lambda deploy-serverless --framework net8.0` or by `"framework": "net8.0"` in `aws-lambda-tools-defaults.json`. I didn't think we have access to that from the source generator, so didn't want to guess.